### ISSUE TITLE
Support for simple wilcoxon / t-test through the same interface

### ISSUE THREAD

**Issue by grst** [scverse/multi-condition-comparisions#25]
### Description of feature

At least on pseudobulked data, wilxocon and t-test can still be viable options for quick comparisons. It would be great to support them through the same interface. 

A t-test is equivalent to a linear model with no covariates. A wilcoxon test is (somewhat (?)) similar to a robust linear model.

**Comment by grst** [comment]
Hi @Zethson and @ilan-gold, 

you brought up if and how this would fit into the API defined for the more complex methods. Even if it doesn't fit perfectly, I still think there is some value to support this interface as it makes it easy to switch between methods. It's also great for benchmarking. 

What we have currently: 

```python
model = SomeModel(adata, design = "~ variable", mask = ..., layer = ...)
model.fit()
model.test_contrasts()
```

For t-test/wilcoxon test

 * The model definition still makes sense, as you need `adata`, `mask` and `layer` and a `design`. However, the `design` doesn't support any covariates in this case. Could just do something like 
    ```python
    variable_to_test = design.replace("~", "").strip()
    if `+` in variable_to_test:
      raise ValueError("Complex designs not supported")
   ```
   
* `fit` doesn't make any sense. But it could still be there and do nothing (wouldn't be necessary to call it, but you could if you are trying different methods)

* `test_contrasts` still makes sense, since you could have comparisons even for a single variable (e.g. `A` vs. `B`, `B` vs. `C`, `A` vs. rest)
